lib/cli.js

```diff
@@ -397,6 +407,21 @@ module.exports = async (argv) => {
 			: path.resolve(process.cwd(), cli.flags.configBasedir);
 	}
 
+	if (cli.flags.globbyOptions) {
+		const value = cli.flags.globbyOptions;
+
+		try {
+			optionsBase.globbyOptions = JSON.parse(value);
+		} catch (_) {
+			process.stderr.write(
+				`Invalid option ${red('"--globby-options"')}. The value ${red(
+					`"${value}"`,
+				)} is not valid JSON.${EOL}`,
+			);
+			process.exit(EXIT_CODE_ERROR); // eslint-disable-line no-process-exit
+		}
+	}
```

Comment:
[suggest] I notice just `JSON.parse` is insufficient because a value like `--globby-options "true"` is unintentionally passed. So we need to check whether the parsed value is an object. We have a utility for that purpose: `isPlainObject()`. See below:

https://github.com/stylelint/stylelint/blob/042d1adee8a15f61d4d3fdf5c2e0e38569303942/lib/utils/validateTypes.js#L65

Thus, I suggest the following additionally:

- Extract a function to parse the `--globby-options` value
- Use `isPlainObject()` to check the JSON-parsed value

```js
	if (cli.flags.globbyOptions) {
		try {
			optionsBase.globbyOptions = await parseGlobbyOptions(cli.flags.globbyOptions);
		} catch (error) {
			if (typeof error === 'string') {
				process.stderr.write(`${error}${EOL}`);
				process.exitCode = EXIT_CODE_ERROR;

				return;
			}

			throw error;
		}
	}

// ...

/**
 * @param {string} value
 * @returns {Promise<Record<string, unknown>>}
 */
function parseGlobbyOptions(value) {
	const errorMessage = () =>
		`Invalid option ${red('"--globby-options"')}.` +
		` The value ${red(`"${value}"`)} is not valid JSON object.`;

	let options;

	try {
		options = JSON.parse(value);
	} catch (_) {
		return Promise.reject(errorMessage());
	}

	if (isPlainObject(options)) {
		return Promise.resolve(options);
	}

	return Promise.reject(errorMessage());
}
```

What do you think?

Reply:
> [suggest] I notice just `JSON.parse` is insufficient because a value like `--globby-options "true"` is unintentionally passed. So we need to check whether the parsed value is an object. We have a utility for that purpose: `isPlainObject()`. See below:
> 
> https://github.com/stylelint/stylelint/blob/042d1adee8a15f61d4d3fdf5c2e0e38569303942/lib/utils/validateTypes.js#L65
> 
> Thus, I suggest the following additionally:
> 
> * Extract a function to parse the `--globby-options` value
> * Use `isPlainObject()` to check the JSON-parsed value
> 
> ```js
> 	if (cli.flags.globbyOptions) {
> 		try {
> 			optionsBase.globbyOptions = await parseGlobbyOptions(cli.flags.globbyOptions);
> 		} catch (error) {
> 			if (typeof error === 'string') {
> 				process.stderr.write(`${error}${EOL}`);
> 				process.exitCode = EXIT_CODE_ERROR;
> 
> 				return;
> 			}
> 
> 			throw error;
> 		}
> 	}
> 
> // ...
> 
> /**
>  * @param {string} value
>  * @returns {Promise<Record<string, unknown>>}
>  */
> function parseGlobbyOptions(value) {
> 	const errorMessage = () =>
> 		`Invalid option ${red('"--globby-options"')}.` +
> 		` The value ${red(`"${value}"`)} is not valid JSON object.`;
> 
> 	let options;
> 
> 	try {
> 		options = JSON.parse(value);
> 	} catch (_) {
> 		return Promise.reject(errorMessage());
> 	}
> 
> 	if (isPlainObject(options)) {
> 		return Promise.resolve(options);
> 	}
> 
> 	return Promise.reject(errorMessage());
> }
> ```
> 
> What do you think?

yeah that's what i trying to point out in last [discussion](https://github.com/stylelint/stylelint/pull/6437#discussion_r1010530350) . 

This looks clean and more transparent approach! Will update it :)